lib/AST/Type.cpp

```diff
@@ -1342,19 +1342,33 @@ Type TypeBase::removeArgumentLabels(unsigned numArgumentLabels) {
   return FunctionType::get(unlabeledParams, result, fnType->getExtInfo());
 }
 
+Type TypeBase::eraseDynamicSelfType() {
+  if (!hasDynamicSelfType())
+    return Type(this);
+
+  return Type(this).transformWithPosition(
+      TypePosition::Covariant,
+      [&](TypeBase *t, TypePosition pos) -> std::optional<Type> {
+        if (auto *selfTy = dyn_cast<DynamicSelfType>(t)) {
+          return selfTy->getSelfType();
+        }
+        return std::nullopt;
```

Comment:
Shouldn't this be checking for `TypePosition::Covariant`, same as `replaceDynamicSelfType`?

Reply:
Nice catch!